### DISCUSSION TITLE
Test how much data is needed to predict an environment

### DISCUSSION THREAD

**Discussion by Astidac** [sdmTMB/sdmTMB#252] [General]
Hi, 
I have sampled data in an area for 5 years, and want to test how much a predictive model improves per year of data added. The idea is that the first few years are likely to matter more, and that the need for additional years reach diminishing returns at some point. In doing so, I want to use the same model parameters, but different time spans: 2023 is explained by only 2022, 2021-2022, 2020-20202 and 2019-2022. I've used the code below to test how well 2019-2022 data together predicts 2023 , but when I run the model on the 2023-data I get log.lik = 0. 

Could you help me understand this, and let me know if this the best way to preform my test.
Thank you!

##__Data:
test_year <- 2023
train_data <- dati2[!dati2$Year == 2023, ] #2019-2022
test_data <- dati2[dati2$Year == 2023,]

clust <- rep(1, nrow(train_data))
clust[which(train_data$year < test_year)] <- 2
##__Mesh:
bound.outer = 10
max.edge=0.95

##__Mesh:
meshy = inla.mesh.2d(loc=cbind(train_data$X, train_data$Y),
                     max.edge = c(1,20)*max.edge,
                     cutoff =5, 
                     offset = c(max.edge, bound.outer))

meshy <- make_mesh(train_data, c("X", "Y"), mesh = meshy)

##__Model
set.seed(1) 
model <- sdmTMB_cv(log_NASC ~ Distgroup * poly(Week, 2) +  Month+ poly(Mean_depth_int,2),
                   data = dati2,
                   mesh = mesh,
                   family   = gaussian(link = "identity"),   
                   k_folds = 8,
                   control = sdmTMBcontrol(newton_loops = 1)
) 

model$sum_loglik
#[1] -58911.16

model$elpd 
#[1] -1.449242

#__Test data:
log_lik <- sum(model$data$cv_loglik[model$data$year == test_year])
log_lik
#[1] 0

**Comment by ericward-noaa** [comment]
Before dealing with the log likelihood being zero, I think there's some funny things going on in how you're setting up the data. You have `dati2` passed to the function, and you're not specifying the clusters a priori (`k_folds = 8` means that 8 folds are being chosen randomly across all locations / years of data). The code's also not using `clust` or `meshy`. Instead, I'd
- create a new variable `id` in `dati2` that defines 2 clusters (whether or not the year is 2023)
- specify the `fold_ids` argument with this new `id` variable

Try this and then let's look into the log likelihood issue
